### ISSUE TITLE
how to save https response

### ISSUE THREAD

**Issue by ddatsh** [neerolyte/repoproxy#6]

**Comment by neerolyte** [comment]
HTTPS is beyond scope for now, it'll be the sort of thing I can worry about if I get the time to go back and rejig the storage format to support non-200s better (#2).

The basic problem with HTTPS is that to be able to cache it as an intermediate proxy you have to break the entire premise of TLS, that means either presenting self signed or completely invalid certs and just hoping that whatever client that is using HTTPS is going to accept them (I'd not trust any client that does this) or you have to put a CA in the intermediate proxy (repoproxy in this case) and add an entire management process around distributing that CA to all your clients and a lot of code to repoproxy to manage both autosigning certs generated on the fly, checking the validity of upstream certs (and injecting some kind of 5xx error if they're bad) and tracking all the certs you generate.

Basically it's a lot of work and none of it would work well by default even if done properly, so it's not worth doing until some of the more fundamental issues are resolved.
